### ISSUE TITLE
Doesn't build with 1.8.0

### ISSUE THREAD

**Issue by steveklabnik** [tildeio/helix#3]
``` bash
$ rustc --version
rustc 1.8.0 (db2939409 2016-04-11)
```

errors:

```
src/macros.rs:99:127: 99:138 error: `$ret:ty` is followed by `$body:block`, which is not allowed for `ty` fragments
src/macros.rs:99     { $cls:ident; ($($mimpl:tt)*) ; ($($mdef:tt)*) ; def $name:ident( $self_arg:tt , $($arg:ident : $argty:ty),* ) -> $ret:ty $body:block $($rest:tt)* } => {
                                                                                                                                               ^~~~~~~~~~~
```

Any ideas?

**Comment by d-unseductable** [comment]
@steveklabnik, I had the same issue in Ruru when implementing method definitions.

This problem was solved with rust-lang/rfcs#1494 and rust-lang/rust#32945, but it's not included into 1.8.0. Thus I decided to use `ident` instead of `ty` temporarily, until 1.9 is released

**Comment by dakull** [comment]
@steveklabnik  @d-unseductable made those changes in `src/macros.rs` and it now complies just fine:

```rust
    { $cls:ident; ($($mimpl:tt)*) ; ($($mdef:tt)*) ; def $name:ident( $self_arg:tt , $($arg:ident : $argty:ident),* ) -> $ret:ident $body:block $($rest:tt)* } => {
        class_definition! { $cls; ($($mimpl)*) ; ($($mdef)*) ; defn $name ; $self_arg ; ($($arg : $argty),*) ; $body ; $ret ; $($rest)*  }
    };

    { $cls:ident; ($($mimpl:tt)*) ; ($($mdef:tt)*) ; def $name:ident( $self_arg:tt , $($arg:ident : $argty:ident),* ) $body:block $($rest:tt)* } => {
        class_definition! { $cls; ($($mimpl)*) ; ($($mdef)*) ; defn $name ; $self_arg ; ($($arg : $argty),*) ; $body ; () ; $($rest)*  }
    };
```

also the examples and specs are running just fine

**Comment by chancancode** [comment]
👍 thank you for investigating! Mind opening a PR with those changes to the macros (or just update the README to require nightly for now)?

**Comment by chancancode** [comment]
I had to revert the fix because it doesn't seem to work with types like `&[usize]` in the membership example. It doesn't look like this was fixed in 1.9, but it seems to work in beta. So I think we will need to require 1.10, is that correct?

**Comment by steveklabnik** [comment]
Yeah, beta will be 1.10.
